### PR TITLE
Added permanova functions (#59)

### PR DESCRIPTION
**Pull request by sdhutchins**
## Additions
- Added `permanova` function.
- Added function to plot `permanova$top_coefficients`
- Added tests for `permanova` and `top_coefficients_barplot`
- Updated roxygen for `create_pub_table`.

## Deprecations
- Deprecated `save` paramenter of plotting functions.

### CODE REVIEW

File: R/barplot.R [hunk 1]
```diff
@@ -20,10 +19,10 @@ melt_metacoder <- function(obj) {
 #' @title Transform Metacoder Dataframe
 #' @description Transform the dataframe abundance values to percent 100.
 #' @param melted_df A "melted" dataframe from the metacoder object's data.
-#' @param tax_level The taxonomic level.
 #' @return Returns a transformed dataframe.
 #' @importFrom dplyr filter group_by summarize mutate enquo quo_name
 #' @importFrom stats na.omit
+#' @rdname stacked_barplot
 transform_metacoder <- function(melted_df, tax_level) {
```
Comment (grabear) [inline R/barplot.R]:
```suggestion
melted_proportions <- function(melted_df, tax_level) {
```

File: R/barplot.R [hunk 2]
```diff
@@ -1,10 +1,9 @@
 #' @title Melt Metacoder Object
 #' @description Melt the metacoder or phyloseq tables into a dataframe.
-#' @param obj An object to be converted to a metacoder object with \code{\link[MicrobiomeR]{object_handler}}.
 #' @return Returns a melted dataframe.
-#' @rdname melt_metacoder
-#' @importFrom  dplyr right_join setdiff
-#' @importFrom  tidyr gather_
+#' @importFrom dplyr right_join setdiff
```
Comment (grabear) [inline R/barplot.R]:
```suggestion
#' @family Formatting
#' @importFrom dplyr right_join setdiff
```

File: R/barplot.R [hunk 3]
```diff
@@ -20,10 +19,10 @@ melt_metacoder <- function(obj) {
 #' @title Transform Metacoder Dataframe
 #' @description Transform the dataframe abundance values to percent 100.
 #' @param melted_df A "melted" dataframe from the metacoder object's data.
-#' @param tax_level The taxonomic level.
 #' @return Returns a transformed dataframe.
 #' @importFrom dplyr filter group_by summarize mutate enquo quo_name
 #' @importFrom stats na.omit
```
Comment (grabear) [inline R/barplot.R]:
```suggestion
#' @importFrom stats na.omit
#' @family Other Filtering
```

Comment (grabear) [inline R/barplot.R]:
```suggestion
#' @importFrom stats na.omit
#' @family Data Manipulators
```

**Comment by sdhutchins** [comment]
@grabear were these the only review remarks? I know that @github was glitchy on Friday.

**Comment by grabear** [comment]
> @grabear were these the only review remarks? I know that @github was glitchy on Friday.

Yes!  I did it twice, because it didn't show up at first.  When the glitches were resolved I fixed everything.

File: R/barplot.R (hunk 1 above)
Comment (sdhutchins) [reply R/barplot.R]: quick question here...

Would you like me to change this function's name or the parameter to `melted_proportions`

```r
#' @title Transform Metacoder Dataframe
#' @description Transform the dataframe abundance values to percent 100.
#' @param melted_df A "melted" dataframe from the metacoder object's data.
#' @return Returns a transformed dataframe.
#' @importFrom dplyr filter group_by summarize mutate enquo quo_name
#' @importFrom stats na.omit
#' @family Data Manipulators
#' @rdname stacked_barplot
transform_metacoder <- function(melted_df, tax_level) {
  t <- dplyr::enquo(tax_level)
  tax_level.abund <- paste0(dplyr::quo_name(t), ".Abundance")

  melted_df %>%
    dplyr::group_by(SampleID, !!sym(tax_level)) %>%
    dplyr::filter(Abundance > 0) %>%
    dplyr::summarize(!!tax_level.abund := sum(as.numeric(Abundance)), TreatmentGroup = dplyr::first(TreatmentGroup)) %>%
    stats::na.omit() %>%
    dplyr::mutate(Relative.Abundance = 100 * !!sym(tax_level.abund) / sum(!!sym(tax_level.abund)))
}
```

Comment (sdhutchins) [reply R/barplot.R]: Awesome! I'll check it out and update it and the documentation for it.

What about `melt_proportions`?

Comment (sdhutchins) [reply R/barplot.R]: Or `convert_proportions`?

Comment (sdhutchins) [reply R/barplot.R]: See https://github.com/vallenderlab/MicrobiomeR/pull/59/commits/b4391d059889aff6ead1dfcb040af469deea692f

**Comment by sdhutchins** [comment]
@grabear No problem! :sunglasses:

**Pull request merged**